### PR TITLE
chore: generation paths are fully dynamic & supports --flat for compo…

### PR DESCRIPTION
**Pull request by Brocco**
…nents/directives/pipes/services

This is a step in aligning with the style guide and will provide the flexibility needed
Related to #316

### CODE REVIEW

**Comment by filipesilva** [comment]
Nice!

File: addon/ng2/blueprints/route-config/index.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 
 var imports, routeDefinitions;
 
 module.exports = {
   description: 'Registers the route with the router.',
 
-  locals: function (options) {
-    return generateLocals.call(this, options);
+  normalizeEntityName: function () {
+    var parsedPath = dynamicPathParser(this.project, 'ignore');
```
Comment (filipesilva) [inline addon/ng2/blueprints/route-config/index.js]: Is the entity name here supposed to be `ignore`?

Comment (Brocco) [inline addon/ng2/blueprints/route-config/index.js]: There are no dynamically named items in this blueprint, so I just needed some text to pass in, route-config just generates a `route-config.ts` file

Comment (filipesilva) [inline addon/ng2/blueprints/route-config/index.js]: Ah ok. I suppose that is being postponed until we have more info regarding the style guide, yes?

Comment (Brocco) [inline addon/ng2/blueprints/route-config/index.js]: Yes, what I've been adding is to put the underlining support in to allow the blueprints to align with the style guide

**Pull request closed without merging**